tests/ops/integration_tests/saas/test_square_task.py

```diff
@@ -101,3 +103,107 @@ async def test_square_access_request_task(
         min_size=1,
         keys=["id", "location_id", "customer_id", "state"],
     )
+
+
+@pytest.mark.integration_saas
+@pytest.mark.integration_square
+@pytest.mark.asyncio
+async def test_square_erasure_request_task(
+    db,
+    policy,
+    erasure_policy_string_rewrite,
+    square_connection_config,
+    square_dataset_config,
+    square_erasure_identity_email,
+    square_erasure_data,
+) -> None:
+    """Full erasure request based on the Square SaaS config"""
+
+    privacy_request = PrivacyRequest(
+        id=f"test_square_erasure_request_task_{random.randint(0, 1000)}"
+    )
+    identity_kwargs = {"email": square_erasure_identity_email}
+    identity = Identity(**identity_kwargs)
+    privacy_request.cache_identity(identity)
+
+    dataset_name = square_connection_config.get_saas_config().fides_key
+    merged_graph = square_dataset_config.get_graph()
+    graph = DatasetGraph(merged_graph)
+    v = await graph_task.run_access_request(
+        privacy_request,
+        policy,
+        graph,
+        [square_connection_config],
+        identity_kwargs,
+        db,
+    )
+
+    assert_rows_match(
+        v[f"{dataset_name}:customer"],
+        min_size=1,
+        keys=[
+            "id",
+            "created_at",
+            "updated_at",
+            "given_name",
+            "family_name",
+            "nickname",
+            "email_address",
+            "address",
+            "phone_number",
+            "company_name",
+            "preferences",
+            "creation_source",
+            "birthday",
+            "segment_ids",
+            "version",
+        ],
+    )
+    # verify we only returned data for our identity email
+    for customer in v[f"{dataset_name}:customer"]:
+        assert customer["email_address"] == square_erasure_identity_email
+
+    assert_rows_match(
+        v[f"{dataset_name}:locations"],
+        min_size=1,
+        keys=[
+            "id",
+            "name",
+            "address",
+            "timezone",
+            "capabilities",
+            "status",
+            "created_at",
+            "merchant_id",
+            "country",
+            "language_code",
+            "currency",
+            "business_name",
+            "type",
+            "business_hours",
+            "mcc",
+        ],
+    )
+    assert_rows_match(
+        v[f"{dataset_name}:orders"],
+        min_size=1,
+        keys=["id", "location_id", "customer_id", "state"],
+    )
+    temp_masking = config.execution.masking_strict
+    config.execution.masking_strict = True
+
+    x = await graph_task.run_erasure(
+        privacy_request,
+        erasure_policy_string_rewrite,
+        graph,
+        [square_connection_config],
+        identity_kwargs,
+        get_cached_data_for_erasures(privacy_request.id),
+        db,
+    )
+    assert x == {
+        f"{dataset_name}:customer": 0,
```

Comment:
I would expect this to be `1` to reflect that the customer was updated, all 0s means the erasure didn't run.